packages/solc-transpiler/src/compiler.ts

```diff
@@ -110,6 +113,53 @@ export const compile = (configJsonString: string, callbacks?: any): string => {
   return formatOutput(res, json)
 }
 
+const getCompiler = (versionString: string): typeof solc => {
+  if (versionString) {
+    const getCompilerString = `
+    function httpsRequest(params, postData) {
+      return new Promise(function(resolve, reject) {
+          var req = https.request(params, function(res) {
+              var body = [];
+              res.on("data", function(chunk) {
+                  body.push(chunk);
+              });
+              res.on("end", function() {
+                  try {
+                      body = Buffer.concat(body).toString();
+                  } catch(e) {
+                      reject(e);
+                  }
+                  resolve(body);
+              });
+          });
+          req.on("error", function(err) {
+              reject(err);
+          });
+          req.end();
+      });
+    }
+
+    async function getSolcVersion(versionString) {
+      const listUrl = "https://ethereum.github.io/solc-bin/bin/list.json";
+      const {releases} = JSON.parse(await httpsRequest(listUrl))
+      const solcUrl = "https://ethereum.github.io/solc-bin/bin/" + releases[versionString];
+      return await httpsRequest(solcUrl);
+    }
+
+    (async () => {
+      await process.stdout.write(await getSolcVersion("${versionString}"));
+    })();
+    `
```

Comment:
Why do we have to use inline executed code btw? Just curious lol seems like a very cool solution

Reply:
yeah, it's wackyness!

Since Truffle expects `compile` to be a synchronous function we can't do anything that's asynchronous in that function. The only way to make a web request "synchronously" in node is to fire up a whole other node process with `execSync` 🙃 

https://stackoverflow.com/a/28394895